### PR TITLE
Added beforeShow and beforeHide events (Issue #84)

### PR DESCRIPTION
**Pull request by leoquijano**

### CODE REVIEW

File: src/easyzoom.js [hunk 1]
```diff
@@ -73,6 +79,8 @@
         var w1, h1, w2, h2;
         var self = this;
 
+        if (!this.opts.beforeShow.call(this)) return;
```
Comment (i-like-robots) [inline src/easyzoom.js]: The default value is delegated to `$.noop` which will return `undefined` which will make this return early.

File: src/easyzoom.js [hunk 2]
```diff
@@ -238,6 +246,7 @@
      */
     EasyZoom.prototype.hide = function() {
         if (!this.isOpen) return;
+        if (!this.opts.beforeHide.call(this)) return;
```
Comment (i-like-robots) [inline src/easyzoom.js]: Same as above. Perhaps the default value for the callbacks should be `null` so we can test whether or not to call the function?

**Comment by leoquijano** [comment]
Thanks for the feedback @i-like-robots . I hadn't much change to test.

I changed the code now to default to null. I also checks for ```if(beforeXX)``` to allow ```false``` to be used as a default as well.

**Comment by i-like-robots** [comment]
Nice one!

I've thought of a second option, maybe the function should explicitly return `false`. The default options could remain `$.noop` and the condition changed to `this.opts.beforeShow.call(this) === false`.

What I like about this is that I think it will cause fewer mistakes for developers using the feature.

What do you think?

**Comment by leoquijano** [comment]
Like this?

**Comment by i-like-robots** [comment]
Exactly what I had in mind, does this way still fit your needs?

**Comment by leoquijano** [comment]
Sure it does. I just have to adjust my code. No biggie ;)

**Comment by i-like-robots** [comment]
Thanks! I will update the docs before releasing

**Pull request merged**